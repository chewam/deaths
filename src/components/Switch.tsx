import { useIntl } from "react-intl"
import { useEffect, useState } from "react"
import { useTheme } from "@/services/themes"
// import { FaMoon, FaSun } from "react-icons/fa"

const Switch = (): JSX.Element | null => {
  const { theme, setTheme } = useTheme()
  const { formatMessage: fm } = useIntl()
  const [mounted, setMounted] = useState(false)

  useEffect(() => setMounted(true), [])

  if (!mounted) return null

  return (
    <div className="switch">
      {theme === "dark" ? (
        // <FaSun
        //   size={11}
        //   title={fm({ id: "light mode" })}
        //   onClick={() => setTheme("light")}
        // />
        <i className="ri-sun-line"></i>
      ) : (
        // <FaMoon
        //   size={11}
        //   title={fm({ id: "dark mode" })}
        //   onClick={() => setTheme("dark")}
        // />
        <i className="ri-moon-line"></i>
      )}
    </div>
  )
}

export default Switch
