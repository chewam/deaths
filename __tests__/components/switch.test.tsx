/**
 * @jest-environment jsdom
 */
import messages from "@/lang/fr.json"
import Switch from "@/components/Switch"
import { IntlProvider } from "react-intl"
import { ThemeProvider } from "next-themes"
import { render } from "@testing-library/react"

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

describe("Switch", () => {
  test("should create a Switch in light mode", () => {
    const { asFragment } = render(
      <IntlProvider locale="fr" messages={messages}>
        <ThemeProvider defaultTheme="light">
          <Switch />
        </ThemeProvider>
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should create a Switch in dark mode", () => {
    const { asFragment } = render(
      <IntlProvider locale="fr" messages={messages}>
        <ThemeProvider
          defaultTheme="dark"
          forcedTheme="dark"
          themes={["light", "dark"]}
        >
          <Switch />
        </ThemeProvider>
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
