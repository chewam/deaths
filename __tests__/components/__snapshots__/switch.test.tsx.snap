// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should create a Switch in dark mode 1`] = `
<DocumentFragment>
  <div
    class="switch"
  >
    <i
      class="ri-sun-line"
    />
  </div>
</DocumentFragment>
`;

exports[`Switch should create a Switch in light mode 1`] = `
<DocumentFragment>
  <div
    class="switch"
  >
    <i
      class="ri-moon-line"
    />
  </div>
</DocumentFragment>
`;
